src/mngt/ocf_mngt_cache.c

```diff
@@ -960,14 +957,44 @@ static void _ocf_mngt_attach_load_properties(ocf_pipeline_t pipeline,
 	context->metadata.dirty_flushed = DIRTY_FLUSHED;
 	context->metadata.line_size = context->cfg.cache_line_size;
 
+	ocf_pipeline_next(pipeline);
+}
+
+static void _ocf_mngt_attach_read_properties(ocf_pipeline_t pipeline,
+		void *priv, ocf_pipeline_arg_t arg)
+{
+	struct ocf_cache_attach_context *context = priv;
+	ocf_cache_t cache = context->cache;
+
 	if (context->cfg.force)
-		OCF_PL_NEXT_RET(context->pipeline);
+		OCF_PL_NEXT_RET(pipeline);
+
+	if (cache->metadata.is_volatile)
+		OCF_PL_NEXT_RET(pipeline);
```

Comment:
We must check for pre-existing metadata even if starting in volatile mode. In volatile mode we will not write our own superblock to the device, but we should check if there already is one.